src/main/kotlin/no/nav/syfo/identhendelse/database/IdenthendelseQuery.kt

```diff
@@ -0,0 +1,123 @@
+            inactiveIdenter.forEach { inactiveIdent ->
+                it.setString(1, nyPersonident.value)
+                it.setString(2, inactiveIdent.value)
+                updatedRows += it.executeUpdate()
+            }
+        }
+        connection.commit()
+    }
+    return updatedRows
+}
+
+const val queryUpdateKandidatEndring =
+    """
+        UPDATE DIALOGMOTEKANDIDAT_ENDRING
+        SET personident = ?
+        WHERE personident = ?
+    """
+
+fun DatabaseInterface.updateKandidatEndring(nyPersonident: PersonIdentNumber, inactiveIdenter: List<PersonIdentNumber>): Int {
+    var updatedRows = 0
+    this.connection.use { connection ->
+        connection.prepareStatement(queryUpdateKandidatEndring).use {
+            inactiveIdenter.forEach { inactiveIdent ->
+                it.setString(1, nyPersonident.value)
+                it.setString(2, inactiveIdent.value)
+                updatedRows += it.executeUpdate()
+            }
+        }
+        connection.commit()
+    }
+    return updatedRows
+}
+
+const val queryUpdateUnntak =
+    """
+        UPDATE UNNTAK
+        SET personident = ?
+        WHERE personident = ?
+    """
+
+fun DatabaseInterface.updateUnntak(nyPersonident: PersonIdentNumber, inactiveIdenter: List<PersonIdentNumber>): Int {
+    var updatedRows = 0
+    this.connection.use { connection ->
+        connection.prepareStatement(queryUpdateUnntak).use {
+            inactiveIdenter.forEach { inactiveIdent ->
+                it.setString(1, nyPersonident.value)
+                it.setString(2, inactiveIdent.value)
+                updatedRows += it.executeUpdate()
+            }
+        }
+        connection.commit()
+    }
+    return updatedRows
+}
+
+const val queryUpdateDialogmoteStatus =
+    """
+        UPDATE DIALOGMOTESTATUS
+        SET personident = ?
+        WHERE personident = ?
+    """
+
+fun DatabaseInterface.updateDialogmoteStatus(nyPersonident: PersonIdentNumber, oldIdentList: List<PersonIdentNumber>): Int {
+    var updatedRows = 0
+    this.connection.use { connection ->
+        connection.prepareStatement(queryUpdateDialogmoteStatus).use {
+            oldIdentList.forEach { oldPersonident ->
+                it.setString(1, nyPersonident.value)
+                it.setString(2, oldPersonident.value)
+                updatedRows += it.executeUpdate()
+            }
+        }
+        connection.commit()
+    }
+    return updatedRows
+}
+
+const val queryFindIdentOccurence =
+    """
+        SELECT COUNT(*)
+        FROM (
+            SELECT personident FROM DIALOGMOTESTATUS
+            UNION ALL
+            SELECT personident FROM DIALOGMOTEKANDIDAT_STOPPUNKT
+            UNION ALL
+            SELECT personident FROM DIALOGMOTEKANDIDAT_ENDRING
+            UNION ALL
+            SELECT personident FROM UNNTAK
+        ) identer
+        WHERE personident = ?
+    """
+
+fun DatabaseInterface.getIdentOccurenceCount(
```

Comment:
Er både occurence og count nødvendig, det blir litt som å si: bilTilfelleAntall? Occurence kan også betyr hendelse, så det kan bli misforstått. Tror også det er fint å få med det vi teller, da får vi et felles begrep på alle tabellene som slås sammen, som vel her er Hendelser? Noen forslag: countIdentHendelser, getIdentHendelseCount

Reply:
Ja, occurence er kanskje litt rart, jeg har tenkt mer på det som en mål på antall treff i databasen ("how often does it occur in the database", hvor ofte forekommer/skjer det).

Det vi teller er jo egentlig ikke identhendelser, i den forstand vi (til nå) har brukt _hendelser_ om endringer som skjer på identer fra PDL. Her teller vi jo egentlig bare hvor mange av de fire tabellene som har en viss ident (og trenger sådan ikke å være knyttet til om det er en ny eller gammel ident, eller hendelse fra PDL). Så vet ikke om hendelser blir riktig, i så fall ville jeg tenkt at en `getIdentHendelseCount()`-funksjon ville telt antall identhendelser som kommer på topicen fra PDL.

Vanskelig med navn, ja 🤔  `getIdentCount`, `countIdentOccurenceAcrossTables`?